subprojects/base-services/src/test/groovy/org/gradle/internal/file/DefaultFileHierarchySetTest.groovy

```diff
@@ -103,6 +104,25 @@ class DefaultFileHierarchySetTest extends Specification {
         set.contains(tmpDir.file("any"))
     }
 
+    @Requires(TestPrecondition.WINDOWS)
+    @Unroll
+    def 'can handle more root dirs'() {
```

Comment:
Can you add some tests for completely different paths on Unix as well? Something like `/var/log` and `/home/user`.

Moreover, it would be good to add some tests for creating a new `DefaultFileHierarchySet` with three elements, aka adding an element when the prefix is already empty.